### PR TITLE
feat(client): add ArtifactsAggregator for assembling artifacts from stream

### LINKED ISSUE #1030: Feat]: Client-side utility for assembling artifacts from update events
### Is your feature request related to a problem? Please describe.

1.0 release removed `ClientTaskManager` (see #734) and changed client API in a way that it now closely mirrors server API and uses [`StreamResposne`](https://a2a-protocol.org/latest/specification/#323-stream-response) as a return type.

#734 mentions a possible helper for reconstructing artifacts specifically, this issue is to track it.

### Describe the solution you'd like

As stated in #734 maintaining the whole `Task` state locally can result in divergence with server state and should not be relied on by default (especially due to history behavior specification in 1.0, #734 has more details).

For artifacts though the logic is deterministic and can be replicated locally if one wants to maintain full artifact state by interpreting `append` and `lastChunk` flags. Client API should not be changed here, new helper should operate on a stream of `StreamResponse`.

### PR DESCRIPTION
**Pull request by Alex-Resch**
## Description

  Adds `ArtifactsAggregator`, a client-side utility for reconstructing complete
  `Artifact` objects from a stream of `StreamResponse` events, as proposed in #734
  and tracked in #1030.

  The helper interprets the `append` and `last_chunk` flags of
  `TaskArtifactUpdateEvent` to correctly assemble chunked and interleaved artifacts
  without requiring changes to the existing client API.

  ### Usage

  ```python
  from a2a.client import ArtifactsAggregator

  # Assemble a single artifact by ID
  aggregator = ArtifactsAggregator.from_stream(client.send_message(request))
  artifact = await aggregator.get_artifact('my-artifact-id')

  # Assemble all artifacts
  aggregator = ArtifactsAggregator.from_stream(client.send_message(request))
  artifacts = await aggregator.get_all_artifacts()
  ```

  ### Changes
  
  - `src/a2a/client/artifact_aggregator.py` — new ArtifactsAggregator class
  - `src/a2a/client/__init__.py` — export ArtifactsAggregator
  - `docs/migrations/v1_0/README.md` — document usage under "Client: Send Message"
  - `tests/client/test_artifact_aggregator.py` — unit tests for all methods and edge cases

  Fixes #1030 🦕

### CODE REVIEW

File: src/a2a/client/artifact_aggregator.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        """Create an ArtifactsAggregator from an async stream of StreamResponse events.
+
+        Args:
+            stream: An async iterator of StreamResponse objects, typically obtained
+                from BaseClient.send_message.
+
+        Returns:
+            A new ArtifactsAggregator instance.
+        """
+        return cls(stream)
```
Comment (sokoliva) [inline src/a2a/client/artifact_aggregator.py]: This method duplicates the constructor. Instead of using `artifact_aggregator = ArtifactsAggregator.from_stream(base_client.send_message(request))` we could directly use `artifact_aggregator = ArtifactsAggregator(base_client.send_message(request))`.

Consider removing it or rethinking the design of ArtifactsAggregator.

**Review by sokoliva — CHANGES_REQUESTED** [review]
